go/tasks/pluginmachinery/internal/webapi/allocation_token.go

```diff
@@ -0,0 +1,95 @@
+package webapi
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	clock2 "k8s.io/utils/clock"
+
+	"github.com/lyft/flytestdlib/logger"
+
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/webapi"
+
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/core"
+)
+
+var (
+	clock clock2.Clock
+)
+
+func SetClockForTest(clck clock2.Clock) {
+	clock = clck
+}
+
+func init() {
+	clock = clock2.RealClock{}
+}
+
+func allocateToken(ctx context.Context, p webapi.AsyncPlugin, tCtx core.TaskExecutionContext, state *State, metrics Metrics) (
```

Comment:
should these be bound to a struct? In that case clock could be a property of the struct. The module var called clock looks a little off IMO (subjective). I am basically vary of too many inits